test/framework/template.go

```diff
@@ -0,0 +1,89 @@
+package framework
+
+import (
+	"context"
+	"os"
+	"testing"
+
+	anywherev1 "github.com/aws/eks-anywhere/pkg/api/v1alpha1"
+	releasev1 "github.com/aws/eks-anywhere/release/api/v1alpha1"
+)
+
+// ProviderTemplateNameGenerator is an interface for getting template name for each provider.
+type ProviderTemplateNameGenerator interface {
```

Comment:
why does this need to be exported? same question for all the methods
I think this is all internal to the e2e framework, it doesn't need to be exposed